go/worker/lambda/lambdaManager.go

```diff
@@ -206,6 +222,7 @@ func (mgr *LambdaMgr) Cleanup() {
 		slog.Info(fmt.Sprintf("Kill function: %s", f.name))
 		f.Kill()
 	}
+	mgr.mapMutex.Unlock()
```

Comment:
Is it important that we're unlocked for the following cleanup, or could we use the standard "defer unlock" pattern up top?